### PR TITLE
Add more v3 samples

### PR DESCRIPTION
**Pull request by jkutner**
This change adds support for all the samples in the https://github.com/buildpack/samples repo. In particular the new `java-buildpack`.

### CODE REVIEW

**Comment by jkutner** [comment]
I've updated this PR to use `build-image` and `run-image`

File: v3/bin/build [hunk 1]
```diff
@@ -7,25 +7,47 @@ cd $(dirname "${BASH_SOURCE[0]}")/..
...
   >&2 echo "Stack not specified."
   exit 1
 fi
 
-rm -rf "$buildpack_dir"
-mkdir -p "$nodejs_dir"
+rm -rf "$buildpacks_dir"
+mkdir -p "$buildpacks_dir"
 
-tar_compat=$(if tar --version | grep -q "GNU tar" ; then echo '--wildcards'; fi)
```
Comment (sclevine) [inline v3/bin/build]: Could we pull in this refactoring?

File: v3/bin/build [hunk 2]
```diff
@@ -7,25 +7,47 @@ cd $(dirname "${BASH_SOURCE[0]}")/..
...
+
+  version_dir=$(dirname "$latest_dir")/$(cat "$latest_dir/buildpack.toml" | yj -t | jq -r .version)
+  mv "$latest_dir" "$version_dir"
+  ln -s "$(basename "$version_dir")" "$latest_dir"
+
+cat << TOML >> "$buildpacks_dir/order.toml"
+  [[groups.buildpacks]]
+    id = "sh.packs.samples.buildpack.${buildpack}"
+    version = "latest"
+    optional = true
```
Comment (sclevine) [inline v3/bin/build]: The `optional` field lets the group pass even if an individual optional buildpack fails. This is to permit extension buildpacks that may not apply. Essentially:
```
Group 1: newrelic-buildpack (optional) | java-buildpack
```
is a shorter, more efficient version of:
```
Group 1: newrelic-buildpack | java-buildpack
Group 2: java-buildpack
```
Making all buildpacks in a group optional means all buildpacks that individually pass detection would apply. Not sure if this is what we want. If Node.js + Java is common (for Java apps with Node assets), maybe something like:
```toml
[[groups]]
  build-image = "packs/v3:build"
  run-image = "packs/v3:run"
  buildpacks = [
    { id = "io.buildpacks.samples.buildpacks.nodejs, version = "latest", optional = true },
    { id = "io.buildpacks.samples.buildpacks.java, version = "latest" },
  ]
[[groups]]
  build-image = "packs/v3:build"
  run-image = "packs/v3:run"
  buildpacks = [
    { id = "io.buildpacks.samples.buildpacks.nodejs, version = "latest" },
  ]
```

File: v3/bin/build (hunk 1 above)
Comment (jkutner) [reply v3/bin/build]: for sure. i missed that

File: v3/bin/build (hunk 2 above)
Comment (jkutner) [reply v3/bin/build]: I actually think `optional` might be broken. When it's `false`, the whole detection fails when one buildpack fails.

Comment (jkutner) [reply v3/bin/build]: I started with the `order.toml` you have there, and my detect failed because the Node.js detect failed.

Comment (jkutner) [reply v3/bin/build]: wait no, my toml was different.

**Pull request merged**